internal/app/init/main.go

```diff
@@ -48,6 +50,31 @@ func kmsg(prefix string) (*os.File, error) {
 	return out, nil
 }
 
+func container() (err error) {
+	log.Println("preparing container based deploy")
+
+	log.Println("remounting volumes as shared mounts")
+	targets := []string{"/", "/var/lib/kubelet", "/etc/cni"}
+	for _, t := range targets {
+		if err = unix.Mount("", t, "", unix.MS_SHARED, ""); err != nil {
```

Comment:
It may be possible to remove this code if we use bind mounts instead of volumes.
That is a bit of a trade-off, and could have a significant perf hit on docker-for-mac/windows.

ref:
> Bind propagation defaults to rprivate for both bind mounts and volumes. It is only configurable for bind mounts, and only on Linux host machines.
https://docs.docker.com/storage/bind-mounts/#configure-bind-propagation

This solution seems good since we have `CAP_SYS_MOUNT` and `CAP_SYS_ADMIN` from the privileged container.

Why do we re-mount `/` ?

Reply:
I like this idea because then we wouldn't have to handle that in code. On the other hand, I personally have experienced  the performance issue. I'd like to get to a point where we have something like `osctl up` and it spins up the cluster using containers. A bit like KIND. So there is room and time for improvement as we